src/lib/requests/common.js

```diff
@@ -50,23 +53,55 @@ const throwOrJson = async (res) => {
     // TODO: will a 503 or 500 with content-length zero generate an error?
     // or a 404 for that matter?!
 
-    if (res.headers['content-length'] === '0' || res.statusCode === 204) {
+    if (res.headers?.['content-length'] === '0' || res.statusCode === 204) {
         // success but no content, return null
         return null;
     }
     if (res.statusCode < 200 || res.statusCode >= 300) {
         // not a successful request
         throw new HTTPResponseError({
-            msg: `Request returned non-success status code ${res.statusCode}`,
+            msg: `${ERROR_MESSAGES.nonSuccessStatusCode} ${res.statusCode}`,
             res,
         });
     }
 
     return res.data;
 };
 
+const defineAgent = (url) =>  {
+    if (!url?.includes('http')) {
+        throw new Error(ERROR_MESSAGES.noProtocolInUrl);
+    }
+
+    const httpModule = url.startsWith('https') ? https : http;
```

Comment:
What about TLS certs?

Or are we handling this through a proxy?

Reply:
I don't change existing logic of handling TLS certs, I just added possibility to do initial httpS call to hub OIDC endpoint (which is exposed with httpS protocol) to get _access_token_ (and set it to Authorization header of any further requests). The rest of logic remains the same

https://github.com/pm4ml/mcm-client/pull/6/files#diff-166152a9b4beec6af970401bb604007a91f007894bb2e04027b29a7411f89805R41